### DISCUSSION TITLE
Installing Ipopt On Linux Seems To Be Single Threaded

### DISCUSSION THREAD

**Discussion by sthomas1978** [coin-or/Ipopt#640] [General]
Hi All

I am trying to install ipopt with the mumps linear solver as detailed in the installation instructions https://coin-or.github.io/Ipopt/INSTALL.html. I have managed to get ipopt up and running within linux environment and I am are calling ipopt via python pyomo package which runs ok.

What I have noticed is that when we run it via Linux ipopt seems to only run on one CPU core. If we run the same python script on a windows architecture it seems to use all cores available.

Is there something I need to change in order to get the Linux installation to utilise all the CPU cores?

**Comment by svigerske** [comment]
For a build of Ipopt with Mumps, there are two places where parallelization can occur.

1. The linear algebra (Blas/Lapack) package. If your Windows build uses a parallelized Blas/Lapack package, e.g., Intel MKL, then this may use several threads. Blas/Lapack is called by Ipopt itself and by Mumps. Depending on the linear algebra package, there may be different options to specify the number of threads to use. For MKL, two ways are to set the environment variables `OMP_NUM_THREADS` or `MKL_NUM_THREADS`.
2. Mumps nowadays can be built with pthread and/or OpenMP support enabled. If you build MUMPS via the [ThirdParty-Mumps](https://github.com/coin-or-tools/ThirdParty-Mumps/) buildsystem, then both are enabled by default. I don't really know how much multithreading will be enabled in Mumps when it has pthread, but for OpenMP, the MUMPS manual says "parts of the MUMPS code, other than BLAS operations, have been parallelized through the use of OpenMP directives". To enable this, you'll need to set the `OMP_NUM_THREADS` environment variable to the number of threads you want.
